cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapAuthenticationConfiguration.java

```diff
@@ -173,5 +171,11 @@ private static PooledBindAuthenticationHandler getPooledBindAuthenticationHandle
         handler.setAuthenticationControls(new PasswordPolicyControl());
         return handler;
     }
-}
 
+    private static PooledCompareAuthenticationHandler getPooledCustomCompareAuthenticationHandler(final LdapAuthenticationProperties l) {
+        final PooledCompareCustomAttributeAuthenticationHandler handler = new PooledCompareCustomAttributeAuthenticationHandler(
+                Beans.newPooledConnectionFactory(l));
+        handler.setPasswordAttributeName(l.getPrincipalAttributePassword());
+        return handler;
```

Comment:
If the principal attribute password is not defined, could we fallback to the pooledBindAuthnHandler?